src/main/java/org/opensearch/timeseries/model/Config.java

```diff
@@ -0,0 +1,550 @@
+        description = input.readOptionalString();
+        timeField = input.readString();
+        indices = input.readStringList();
+        featureAttributes = input.readList(Feature::new);
+        filterQuery = input.readNamedWriteable(QueryBuilder.class);
+        interval = IntervalTimeConfiguration.readFrom(input);
+        windowDelay = IntervalTimeConfiguration.readFrom(input);
+        shingleSize = input.readInt();
+        schemaVersion = input.readInt();
+        this.categoryFields = input.readOptionalStringList();
+        lastUpdateTime = input.readInstant();
+        if (input.readBoolean()) {
+            this.user = new User(input);
+        } else {
+            user = null;
+        }
+        if (input.readBoolean()) {
+            this.uiMetadata = input.readMap();
+        } else {
+            this.uiMetadata = null;
+        }
+        customResultIndex = input.readOptionalString();
+        if (input.readBoolean()) {
+            this.imputationOption = new ImputationOption(input);
+        } else {
+            this.imputationOption = null;
+        }
+        this.imputer = createImputer();
+    }
+
+    /*
+     * Implicit constructor that be called implicitly when a subtype
+     * needs to call like AnomalyDetector(StreamInput). Otherwise,
+     * we will have compiler error:
+     * "Implicit super constructor Config() is undefined.
+     * Must explicitly invoke another constructor".
+     */
+    public Config() {
+        this.imputer = null;
+    }
+
+    @Override
+    public void writeTo(StreamOutput output) throws IOException {
+        output.writeOptionalString(id);
+        output.writeOptionalLong(version);
+        output.writeString(name);
+        output.writeOptionalString(description);
+        output.writeString(timeField);
+        output.writeStringCollection(indices);
+        output.writeList(featureAttributes);
+        output.writeNamedWriteable(filterQuery);
+        interval.writeTo(output);
+        windowDelay.writeTo(output);
+        output.writeInt(shingleSize);
+        output.writeInt(schemaVersion);
+        output.writeOptionalStringCollection(categoryFields);
+        output.writeInstant(lastUpdateTime);
+        if (user != null) {
+            output.writeBoolean(true); // user exists
+            user.writeTo(output);
+        } else {
+            output.writeBoolean(false); // user does not exist
+        }
+        if (uiMetadata != null) {
+            output.writeBoolean(true);
+            output.writeMap(uiMetadata);
+        } else {
+            output.writeBoolean(false);
+        }
+        output.writeOptionalString(customResultIndex);
+        if (imputationOption != null) {
+            output.writeBoolean(true);
+            imputationOption.writeTo(output);
+        } else {
+            output.writeBoolean(false);
+        }
+    }
+
+    /**
+     * If the given shingle size is null, return default based on the kind of detector;
```

Comment:
nit: This comment mentions we return based on kind of detector but it should both be changed to generic timeseries and also I don't see type of detector or forecaster factored in here, just if there is no custom use default and if custom use custom.

Reply:
yes, fixed.